packages/third-parties/src/google/ga.tsx

```diff
@@ -0,0 +1,55 @@
+'use client'
+// TODO: Evaluate import 'client only'
+import React, { useEffect } from 'react'
+import Script from 'next/script'
+
+declare global {
+  interface Window {
+    dataLayer?: Object[]
+  }
+}
+
+type GAParams = {
+  gaId: string
+}
+
+export function GoogleAnalytics(props: GAParams) {
+  const { gaId } = props
+
+  useEffect(() => {
+    // performance.mark is being used as a feature use signal. While it is traditionally used for performance
+    // benchmarking it is low overhead and thus considered safe to use in production and it is a widely available
+    // existing API.
+
+    performance.mark('mark_feature_usage', {
+      detail: {
+        feature: 'next-third-parties-ga',
+      },
+    })
+  }, [])
+
+  return (
+    <>
+      <Script
+        id="_next-ga-init"
+        dangerouslySetInnerHTML={{
+          __html: `
+          window.dataLayer = window.dataLayer || [];
```

Comment:
Do we need to have custom dataLayer name in GA?

Reply:
in our case, we are a SaSS and we have our own GA but also allow clients to embed their own GA. 
CMIIW we have different dataLayer in this case to avoid messing up the analytics data for both us and the client